sql/src/main/java/io/crate/planner/node/dql/UnionPhase.java

```diff
@@ -0,0 +1,160 @@
+/*
+ * Licensed to Crate under one or more contributor license agreements.
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.  Crate licenses this file
+ * to you under the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.  You may
+ * obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+ * implied.  See the License for the specific language governing
+ * permissions and limitations under the License.
+ *
+ * However, if you have executed another commercial license agreement
+ * with Crate these terms will supersede the license and you may use the
+ * software solely pursuant to the terms of the relevant commercial
+ * agreement.
+ */
+
+package io.crate.planner.node.dql;
+
+import com.google.common.base.MoreObjects;
+import com.google.common.collect.ImmutableSet;
+import com.google.common.collect.Iterables;
+import io.crate.analyze.symbol.Symbols;
+import io.crate.planner.distribution.DistributionInfo;
+import io.crate.planner.distribution.UpstreamPhase;
+import io.crate.planner.node.ExecutionPhaseVisitor;
+import io.crate.planner.projection.Projection;
+import org.elasticsearch.common.io.stream.StreamInput;
+import org.elasticsearch.common.io.stream.StreamOutput;
+
+import java.io.IOException;
+import java.util.*;
+
+import static io.crate.planner.distribution.DistributionInfo.DEFAULT_SAME_NODE;
+
+public class UnionPhase extends AbstractProjectionsPhase implements UpstreamPhase {
+
+    public static final ExecutionPhaseFactory<UnionPhase> FACTORY = new ExecutionPhaseFactory<UnionPhase>() {
+        @Override
+        public UnionPhase create() {
+            return new UnionPhase();
+        }
+    };
+
+    private Collection<String> executionNodes;
+    private Collection<MergePhase> mergePhases;
+
+    public UnionPhase() {
+    }
+
+    public UnionPhase(UUID jobId,
+                      int executionNodeId,
+                      String name,
+                      List<Projection> projections,
+                      Collection<MergePhase> mergePhases,
+                      Collection<String> executionNodes) {
+        super(jobId, executionNodeId, name, projections);
+        assert !mergePhases.isEmpty();
+        if (projections.isEmpty()) {
+            outputTypes = Iterables.getLast(mergePhases).outputTypes();
```

Comment:
last one could be null. anyway it looks a bit odd as due to lastOne usage it looks a bit like mergePhases are chained, I'd suggest add at least a comment that it doesn't matter which mergePhase is used (as long as it is not null)